src/Distribution/Audit.hs

```diff
@@ -194,6 +208,80 @@ osvHandler mkHandle mp =
               ]
         ]
 
+sarifHandler
+  :: MonadUnliftIO m
+  => Codensity IO Handle
+  -> ProjectBaseContext
+  -> [(PackageName, ElaboratedPackageInfoAdvised)]
+  -> m ()
+sarifHandler mkHandle projectBaseContext packageAdvisories = do
+  let projectRoot = distProjectRootDirectory $ distDirLayout projectBaseContext -- TODO(blackheaven): resolve repository root (in GitHub Action, we get the cloned directory, instead of the directory from repository's root)
+  ghcVersion <- liftIO $ T.pack <$> readProcess "ghc" ["--version"] ""
```

Comment:
I have been thinking about this and I think this should be an argument to the command line. It is not a good idea in general to ask the ghc version of the environment, for several reasons, most importantly that this check might run in a different environment than the project is built in, I would even say that this is the regular case. Making this kinda implicit and not making the use explicitly choose the environment is bad, imo. What do you think @blackheaven?

Reply:
Actually, it's a bit more complex, I try to find the version which is compiling `cabal-audit`

Reply:
got it!

Reply:
well but that should be easier, no? you're already accessing static info about the package with `CPP` anyway, afaiu this should just be `__GLASGOW_HASKELL__`? (https://downloads.haskell.org/ghc/latest/docs/users_guide/phases.html#standard-cpp-macros)